app/main/api/handlers/getWallet.ts

```diff
@@ -103,25 +105,32 @@ async function replaceStorage(params: GetWalletParams, system: SubSystems)
 /**
  * search wallet in db
  *
- * @param {{ storage: Storage<Buffer, JsonSerializable, Buffer, Buffer>, id: string }}
- * @returns {Promise<Wallet>}
+ * @param {{ storage: JsonAesLevelStorage, id: string }}
+ * @returns {Promise<JsonSerializable>}
  */
 async function searchWallet(
   { storage, id }: { storage: JsonAesLevelStorage, id: string }
-): Promise<Wallet> {
+): Promise<JsonSerializable> {
   try {
-    const data = await storage.get("wallet")
-
-    try {
-      return asRuntimeType(data, Wallet)
-    } catch (error) {
-      throw getWalletErrors.INVALID_WALLET_TYPE
-    }
+    return await storage.get("wallet")
   } catch (error) {
     throw getWalletErrors.INVALID_PASSWORD
```

Comment:
We could take this chance to rename `INVALID_PASSWORD` as `WRONG_PASSWORD`, as the problem is not about validity (conforming to an expected format) but about it being the right one.